src/vmm/src/lib.rs

```diff
@@ -600,6 +602,50 @@ impl Vmm {
         Ok(cpu_configs)
     }
 
+    /// Adds new vCPUs to VMM.
+    pub fn hotplug_vcpus(&mut self, config: HotplugVcpuConfig) -> Result<(), HotplugVcpuError> {
+        println!("Num vCPUs before hotplug: {}", self.vcpus_handles.len());
+
+        if config.vcpu_count < 1 {
+            return Err(HotplugVcpuError::VcpuCountTooLow);
+        } else if self.vcpus_handles.len() as u8 + config.vcpu_count > MAX_SUPPORTED_VCPUS {
+            return Err(HotplugVcpuError::VcpuCountTooHigh);
+        }
+
+        // Create and start new vcpus
+        let mut vcpus = Vec::with_capacity(config.vcpu_count as usize);
+
+        let start_idx = self.vcpus_handles.len() as u8;
+        for cpu_idx in start_idx..(start_idx + config.vcpu_count) {
+            let exit_evt = self
+                .vcpus_exit_evt
+                .try_clone()
+                .map_err(HotplugVcpuError::EventFd)?;
+            let vcpu =
+                Vcpu::new(cpu_idx, &self.vm, exit_evt).map_err(HotplugVcpuError::VcpuCreate)?;
+            vcpus.push(vcpu);
+        }
+
+        let seccomp_filters = get_empty_filters();
+        self.start_vcpus(
+            vcpus,
+            seccomp_filters
+                .get("vcpu")
+                .unwrap()
+                // .ok_or_else(|| VmmError::SeccompFilters(()))?
+                .clone(),
+        )
+        .map_err(HotplugVcpuError::VcpuStart)?;
+
+        println!("Num vCPUs after hotplug: {}", self.vcpus_handles.len());
```

Comment:
Here, we could have a metric that counts the actual number of hotplugged vCPUs. You could then use that metric as a first, simple integration test (e.g. have an integration test that calls the the hotplug endpoint and then does an assertion on the metric)